app/[locale]/auth/actions.ts

```
@@ -57,28 +60,44 @@ export const signInAction = validatedAction(signInSchema, async (data) => {
     try {
-      await signIn(data.provider, {
-        redirect: false,
+      const authService = authServiceFactory(data.authProvider);
+      
+      const result = await authService.signInWithOAuth(data.provider, {
+        ...data,
+        callbackUrl: data.callbackUrl
       });
-
-      return { success: true };
+      if (result && result.url) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)